go-controller/pkg/allocator/pod/addr_pool.go

```diff
@@ -57,3 +69,62 @@ func calculateSubnetsInfraMACAddresses(subnets []config.CIDRNetworkEntry) map[st
 	}
 	return reservedMACs
 }
+
+// fetchNetworkPods fetch pods in to the network NAD namespaces.
+func (allocator *PodAnnotationAllocator) fetchNetworkPods() ([]*corev1.Pod, error) {
+	var netPods []*corev1.Pod
+	for _, ns := range allocator.netInfo.GetNADNamespaces() {
+		pods, err := allocator.podLister.Pods(ns).List(labels.Everything())
+		if err != nil {
+			return nil, fmt.Errorf("failed to list pods for namespace %q: %v", ns, err)
+		}
+		for _, pod := range pods {
+			if pod == nil {
+				continue
+			}
+			if pod.Status.Phase != corev1.PodRunning || !pod.DeletionTimestamp.IsZero() && len(pod.Finalizers) == 0 {
+				// skip pods who are non-running or about to dispose
+				continue
+			}
+			netPods = append(netPods, pod)
+		}
+	}
+	return netPods, nil
+}
+
+// allocatePodMACs for each given pod it record the given pods MAC addresses in the network pool.
+// In case of a conflict it emits pod event reflecting MAC conflict occurred.
+func (allocator *PodAnnotationAllocator) allocatePodMACs(pods []*corev1.Pod, eventRecorder record.EventRecorder) error {
+	networkName := allocator.netInfo.GetNetworkName()
+	macConflictPods := map[string]string{}
+
+	for _, pod := range pods {
+		podNetworks, err := util.UnmarshalPodAnnotationAllNetworks(pod.Annotations)
+		if err != nil {
+			return fmt.Errorf("failed to unmarshal pod annotation %s/%s: %v", pod.Namespace, pod.Name, err)
+		}
+		for nadName, network := range podNetworks {
+			if network.Role != types.NetworkRoleInfrastructure {
+				// primary UDN network role is infrastructure-lock on primary UDNs only.
+				continue
+			}
+			mac, perr := net.ParseMAC(network.MAC)
+			if perr != nil {
+				return fmt.Errorf("failed to parse pod %s/%s mac address %q: %v", pod.Namespace, pod.Name, network.MAC, perr)
+			}
+
+			ownerID := GetPoolAddressOwner(pod, allocator.netInfo)
+			if allocator.addressPool.IsMACConflict(networkName, mac, ownerID) {
+				macConflictPods[network.MAC] = pod.Namespace + "/" + pod.Name
+				klog.Warningf("%s: %s already allocated in network attachment %s, network-name %s",
+					pool.ErrMACConflict, mac, nadName, allocator.netInfo.GetNetworkName())
+			} else {
+				allocator.addressPool.AddMACToPool(allocator.netInfo.GetNetworkName(), mac, ownerID)
+			}
```

Comment:
as the bot said, I don't think we can ship it like this.

Reply:
Functionality wise, it work as expected. 
Refactoring this as suggested sounds like a cosmetic change that wound not add much value.
Having said that, I dont have strong feeling about changing it.